### ISSUE TITLE
Swift 4 : Declarations from extensions cannot be overridden yet

### ISSUE THREAD

**Issue by KrishnendraPrakash** [dekatotoro/SlideMenuControllerSwift#331]
Integrate in Xcode Version 9.0 beta 3.
Issue: **Declarations from extensions cannot be overridden yet**

Please see screenshot.
<img width="1021" alt="screen shot 2017-07-24 at 1 23 29 pm" src="https://user-images.githubusercontent.com/7422114/28513329-587b051e-7073-11e7-9201-b0d32db84f8e.png">

**Comment by ravi8x** [comment]
I am getting the same problem with XCode 9.0 Beta 5. Any solution yet?

**Comment by abhijithExperion** [comment]
@ravi8x How are you showing menu on the view controllers which is not pushed but instead presented? Any solution?

**Comment by ravi8x** [comment]
@abhijithExperion No Abhijith. I have written my own Slider Menu class with very minimal functionalities.

**Comment by andylamp** [comment]
this is a very easily reproducible "bug"; just check out the current branch and import the library to a project using Swift 4 (e.g. XCode 4 Beta 1-4). Probably it is an easy fix...

**Comment by KrishnendraPrakash** [comment]
@andylamp I am using with pods 
 **pod 'SlideMenuControllerSwift'**
What changes required when using pod with swift 4, XCode - Version 9.0 beta 3 (9M174d).
Thanks

**Comment by andylamp** [comment]
@KrishnendraPrakash the solution, albeit with warnings, for now it seems to be that you have to put the following:

```swift
    @objc public func openLeft() {
      slideMenuController()?.openLeft()
    }
  
  
    @objc public func openRight() {
        slideMenuController()?.openRight()    }
  
  
    @objc public func closeLeft() {
        slideMenuController()?.closeLeft()
    }
  
  
    @objc public func closeRight() {
        slideMenuController()?.closeRight()
    }
```

Specifically, these are located in lines 1055-1075 in SlideMenuController.swift -- they should work but XCode will complain about deprecated functionality... but it's a temporary solution for the time being.

Hope this helps...

**Comment by KeimyPlaza** [comment]
Anybody know if there will be a release for Swift4?

We need the solution by pods installation

**Comment by akshayiosdeveloper** [comment]
Not compatible with swift4

**Comment by andylamp** [comment]
@akshayiosdeveloper as I said, you can alter the source file as indicated in my previous answer and it will run just fine...

**Comment by KeimyPlaza** [comment]
@akshayiosdeveloper you can use too branch of Swift 4 if you use as pod.

https://github.com/dekatotoro/SlideMenuControllerSwift/tree/swift4

@andylamp I hope you MR is comming soon.

**Comment by arun-perk** [comment]
@andylamp  You are right but if we use Pod file, It will auto update when we update our other pod files, right ?

**Comment by andylamp** [comment]
@arun-perk you can use the pods git branch tool in order to have a specific version, this can be done easily like so:

```
platform :ios, '11.0'

target 'YourAppName' do
  use_frameworks!
  # more pods...
    pod 'SlideMenuControllerSwift', :git => 'https://github.com/dekatotoro/SlideMenuControllerSwift.git', :branch => 'swift4''
  # even more pods...
end
```
Hope this helps.

**Comment by jerrypm** [comment]
Update your pod 
![screen shot 2018-02-10 at 23 48 22](https://user-images.githubusercontent.com/15967964/36064395-fe5ca4fa-0ebc-11e8-91b4-f2ed769c4f34.png)
https://guides.cocoapods.org/using/getting-started.html
" sudo gem update cocoapods " in terminal

Note : it work for me 👍
